### PR TITLE
[5.3] Double quote MySQL JSON expressions

### PR DESCRIPTION
**Pull request by themsaid**
Based on MySQL guide: http://dev.mysql.com/doc/refman/5.7/en/json-path-syntax.html

> Names of keys must be double-quoted strings or valid ECMAScript identifiers

Some characters like Hyphens are not allowed in ECMAScript identifiers so we must double quote the parts of the JSON expression for it to work.

This change will convert `meta->name->first-name` to `meta->'$."name"."first-name"'`.

### CODE REVIEW

File: src/Illuminate/Database/Query/Grammars/MySqlGrammar.php [hunk 1]
```diff
@@ -242,7 +242,11 @@ protected function wrapJsonSelector($value)
 
         $field = $this->wrapValue(array_shift($path));
 
-        return $field.'->'.'"$.'.implode('.', $path).'"';
+        $path = collect($path)->map(function ($part) {
+            return '"'.$part.'"';
+        })->toArray();
```
Comment (JosephSilber) [inline src/Illuminate/Database/Query/Grammars/MySqlGrammar.php]: You can call `implode` on the collection:

```php
$path = collect($path)->map(function ($part) {
    return '"'.$part.'"';
})->implode('.');
```

Comment (themsaid) [inline src/Illuminate/Database/Query/Grammars/MySqlGrammar.php]: Thank you 👍

**Comment by taylorotwell** [comment]
Ha, I just merged it and fixed the implode for you :smile:

**Comment by themsaid** [comment]
😅 ok

**Comment by sebestenyb** [comment]
Impressive!
Thank you gentlemen.

Cheers, Sebi

**Pull request closed without merging**